src/include/txn.i

```diff
@@ -863,11 +866,24 @@ __wt_txn_read_upd_list(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE
                 continue;
             break;
         }
-        if (upd_visible == WT_VISIBLE_PREPARE)
-            return (WT_PREPARE_CONFLICT);
     }
     if (upd == NULL)
         return (0);
+
```

Comment:
Moving this condition here let the above loop to continue and doesn't report the prepare conflict for the sessions that didn't pass the `ignore_prepare`. It must be inside that loop.

Reply:
Oh, right